scripts/zones/Riverne-Site_B01/mobs/Bahamut.lua

```diff
@@ -1,15 +1,97 @@
+    mob:addStatusEffect(EFFECT_PROTECT,175,0,1800);
+    mob:addStatusEffect(EFFECT_SHELL,62,0,1800);
+end;
+
+-----------------------------------
+-- onMobFight Action
+-----------------------------------
+
+function onMobFight(mob,target)
+    local MegaFlareQueue = mob:getLocalVar("MegaFlareQueue");
+    local MegaFlareTrigger = mob:getLocalVar("MegaFlareTrigger");
+    local MegaFlareUses = mob:getLocalVar("MegaFlareUses");
+    local FlareWait = mob:getLocalVar("FlareWait")
+    local GigaFlare = mob:getLocalVar("GigaFlare");
+    local tauntShown = mob:getLocalVar("tauntShown");
+    local mobHPP = mob:getHPP();
+    local isBusy = false;
+    if (GetMobAction(mob:getID()) == ACTION_MOBABILITY_START or GetMobAction(mob:getID()) == ACTION_MOBABILITY_USING or GetMobAction(mob:getID()) == ACTION_MOBABILITY_FINISH or GetMobAction(mob:getID()) == ACTION_MAGIC_START or GetMobAction(mob:getID()) == ACTION_MAGIC_CASTING or GetMobAction(mob:getID()) == ACTION_MAGIC_START) then
+        isBusy = true; -- is set to true if Bahamut is in any stage of using a mobskill or casting a spell
+    end;
+    
+    if (mobHPP < 90 and MegaFlareTrigger < 1) then -- if Megaflare hasn't been set to be used this many times, increase the queue of Megaflares. This will allow it to use multiple Megaflares in a row if the HP is decreased quickly enough.
+        mob:setLocalVar("MegaFlareTrigger", 1);
+        mob:setLocalVar("MegaFlareQueue", MegaFlareQueue + 1);
+    elseif (mobHPP < 80 and MegaFlareTrigger < 2) then
+        mob:setLocalVar("MegaFlareTrigger", 2);
+        mob:setLocalVar("MegaFlareQueue", MegaFlareQueue + 1);
+    elseif (mobHPP < 70 and MegaFlareTrigger < 3) then
+        mob:setLocalVar("MegaFlareTrigger", 3);
+        mob:setLocalVar("MegaFlareQueue", MegaFlareQueue + 1);
+    elseif (mobHPP < 60 and MegaFlareTrigger < 4) then
+        mob:setLocalVar("MegaFlareTrigger", 4);
+        mob:setLocalVar("MegaFlareQueue", MegaFlareQueue + 1);
+    elseif (mobHPP < 50 and MegaFlareTrigger < 5) then
+        mob:setLocalVar("MegaFlareTrigger", 5);
+        mob:setLocalVar("MegaFlareQueue", MegaFlareQueue + 1);
+    elseif (mobHPP < 40 and MegaFlareTrigger < 6) then
+        mob:setLocalVar("MegaFlareTrigger", 6);
+        mob:setLocalVar("MegaFlareQueue", MegaFlareQueue + 1);
+    elseif (mobHPP < 30 and MegaFlareTrigger < 7) then
+        mob:setLocalVar("MegaFlareTrigger", 7);
+        mob:setLocalVar("MegaFlareQueue", MegaFlareQueue + 1);
+    elseif (mobHPP < 20 and MegaFlareTrigger < 8) then
+        mob:setLocalVar("MegaFlareTrigger", 8);
+        mob:setLocalVar("MegaFlareQueue", MegaFlareQueue + 1);
+    end;
+
+    if (mob:actionQueueEmpty() == true and isBusy == false) then -- the last check prevents multiple Mega/Gigaflares from being called at the same time.
+        if (MegaFlareQueue > 0) then
+            mob:SetMobAbilityEnabled(false); -- disable all other actions until Megaflare is used successfully

```

Comment:
Seems like all three of these have to be set to true in onMobDespawn, since it persists through to the next fight if it was killed while they were disabled.